### PR TITLE
[DX] Make services private

### PR DESCRIPTION
**Pull request by mpysiak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

### CODE REVIEW

**Review by Wojdylak — DISMISSED** [review]

File: src/Sylius/Bundle/AdminBundle/Resources/config/services/controller.xml [hunk 1]
```diff
@@ -102,7 +102,6 @@
             id="sylius_admin.resource_controller.redirect_handler"
             class="Sylius\Bundle\AdminBundle\Controller\RedirectHandler"
             decorates="sylius.resource_controller.redirect_handler"
-            public="false"
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/Resources/config/services/controller.xml]: Why?

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/mailer.xml [hunk 2]
```diff
@@ -16,18 +16,16 @@
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd"
 >
-    <services>
-        <defaults public="true" />
-
-    <service id="sylius.listener.user_mailer" class="Sylius\Bundle\CoreBundle\EventListener\MailerListener" public="false">
+    <services>       
+        <service id="sylius.listener.user_mailer" class="Sylius\Bundle\CoreBundle\EventListener\MailerListener" public="false">
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/mailer.xml]:
```suggestion
        <service id="sylius.listener.user_mailer" class="Sylius\Bundle\CoreBundle\EventListener\MailerListener">
```

File: src/Sylius/Bundle/AdminBundle/Resources/config/services/controller.xml (hunk 1 above)
Comment (mpysiak) [reply src/Sylius/Bundle/AdminBundle/Resources/config/services/controller.xml]: Probably some conflicts leftover. Fixed

**Review by Wojdylak — APPROVED** [review]

**Review by GSadee — APPROVED** [review]
Add notes to an UPGRADE file and IMO, remove this ADR: https://github.com/Sylius/Sylius/blob/2.0/adr/2022_02_11_declaring_services_in_container.md, but it could be done in a separate PR 🖖🏻

**Pull request merged**